### ISSUE TITLE
Convert plugin settings page to ReactJS + REST API endpoints

### ISSUE THREAD

**Issue by MichelleBlanchette** [PurpleTurtleCreative/completionist#59]
The settings page in the admin has hardly anything on it, yet its code is so complicated and a complete nightmare to work with!!! This should be using ReactJS so that it's much easier to see what's going on and to make easily dynamic.

Also, the rest of the interfaces will be ReactJS, anyways, so this needs to be consistent.

_Note this should be completed before any new plugin options are added, such as mentioned in #37_

**Comment by MichelleBlanchette** [comment]
Going to write a new Issue for this because I have a different plan in mind now that WordPress has evolved quite a bit.
